### PR TITLE
Keep content open hover when focused until escaped or focus moved

### LINKED ISSUE #185198: The hover should not disappear when moving mouse out of it and when it is focused
* hover over a symbol
* click into the hover, focusing it
* move the mouse outside the hover
* the hover should not hide itself

https://github.com/microsoft/vscode/assets/5047891/fc290a6d-9e34-4821-9a2e-fe650bbaf4ae

### PR DESCRIPTION
**Pull request by aiday-mar**
Fixes: https://github.com/microsoft/vscode/issues/185198

### CODE REVIEW

**Review by alexdima — CHANGES_REQUESTED** [review]
I think this should to be done in `ModesHoverController._onEditorMouseMove` somewhere around here -- https://github.com/microsoft/vscode/blob/404280c6233064f98e584347ec189deba4d3e45c/src/vs/editor/contrib/hover/browser/hover.ts#L167-L170

File: src/vs/editor/contrib/hover/browser/contentHover.ts [hunk 1]
```diff
@@ -259,7 +263,9 @@ export class ContentHoverController extends Disposable {
 				return;
 			}
 		}
-
+		if (hoverResult.messages.length === 0 && this._widget.isFocused) {
+			return;
+		}
```
Comment (alexdima) [inline src/vs/editor/contrib/hover/browser/contentHover.ts:268]: Why is this necessary?

**Review by alexdima — COMMENTED** [review]
I'm not sure about the added code in `_withResult`. What does it help with?

File: src/vs/editor/contrib/hover/browser/contentHover.ts (hunk 1 above)
Comment (aiday-mar) [reply src/vs/editor/contrib/hover/browser/contentHover.ts:268]: I added that because sometimes the hoverResult contains a message array of length 0, in which case later in the code, this is detected and we hide the content widget. This means that without this code that was added, when the content hover is focused and the mouse is moved around the editor, sometimes the content widget hides unexpectedly when you quickly move your mouse over some code that generates a hoverResult with message array length 0. I thought it would be better in that case, not to hide the current focused content hover.

Comment (alexdima) [reply src/vs/editor/contrib/hover/browser/contentHover.ts:268]: Thanks for clarifying! This suggests that the proposed change in `_onEditorMouseMove` is not quite correct. The problem is that [this code](https://github.com/microsoft/vscode/blob/ede257b88938d0c5aec83d20ec0c3354e78ae65f/src/vs/editor/contrib/hover/browser/hover.ts#L206) calls `maybeShowAt` which triggers a content widget recomputation and updating of the contents of the widget. But that is wrong, if the widget is focused or if it is being resized that call should not be done. Effectively this means moving the `this._contentWidget?.widget.isResizing` and `this._contentWidget?.isFocused()` calls above the `maybeShowAt` call, maybe somewhere around https://github.com/microsoft/vscode/blob/ede257b88938d0c5aec83d20ec0c3354e78ae65f/src/vs/editor/contrib/hover/browser/hover.ts#L167

Comment (aiday-mar) [reply src/vs/editor/contrib/hover/browser/contentHover.ts:268]: Sounds good, I initially thought it may be a good idea to let recomputation be done when the content widget is focused. The subsequent rendered content hovers would also be focused in that case. I'll change that.

Comment (aiday-mar) [reply src/vs/editor/contrib/hover/browser/contentHover.ts]: I also cleaned up the code a little, we had some of the `isSomethingVisible` or `isSomethingFocused` functions as getters. I made all such similar functions into getters.

**Review by alexdima — APPROVED** [review]
Nice!
